src/main/java/info/skyblond/nekohit/neo/domain/WCABasicInfo.java

```diff
@@ -32,45 +33,61 @@
     public int nextMilestoneIndex;
 
     /**
-     * Indicate if the stake is paid
+     * Status of this WCA. Note: OPEN might become ACTIVE overtime. <br>
+     * 0 - PENDING, the initialized status, only payStake and cancel is allowed.<br>
+     * 1 - OPEN, from PENDING, after pay the stake, ready to operate.<br>
+     * 2 - ACTIVE, from OPEN, after pass the threshold milestone.<br>
+     * 3 - FINISHED, from OPEN or ACTIVE, after creator finish this WCA.<br>
      */
-    public boolean paid;
-
-    /**
-     * If this WCA is accounted and finished.
-     */
-    public boolean finished;
+    public int status;
 
     public WCABasicInfo(
             Hash160 owner, String description,
             int stakePer100Token, int maxTokenSoldCount,
             int milestoneCount, int thresholdIndex, int coolDownInterval, boolean bePublic
     ) throws Exception {
-        require(Hash160.isValid(owner), "Owner address is not a valid address.");
+        require(Hash160.isValid(owner), Messages.INVALID_HASH160);
         this.owner = owner;
-        require(description != null, "Description can be empty, but not null.");
+        require(description != null, Messages.NULL_DESCRIPTION);
         this.description = description;
-        require(stakePer100Token > 0, "The stake amount per 100 token must be positive.");
-        require(maxTokenSoldCount > 0, "The max sell token count must be positive.");
+        require(stakePer100Token > 0, Messages.INVALID_STAKE_RATE);
+        require(maxTokenSoldCount > 0, Messages.INVALID_MAX_SELL_AMOUNT);
         this.stakePer100Token = stakePer100Token;
         this.maxTokenSoldCount = maxTokenSoldCount;
-        require(milestoneCount > 0, "You must have at least 1 milestone.");
+        require(milestoneCount > 0, Messages.INVALID_MILESTONES_COUNT);
         this.milestoneCount = milestoneCount;
         if (thresholdIndex >= 0 && thresholdIndex < this.milestoneCount) {
             this.thresholdIndex = thresholdIndex;
         } else {
-            throw new Exception("Invalid value for thresholdIndex");
+            throw new Exception(Messages.INVALID_THRESHOLD_INDEX);
         }
-        require(coolDownInterval >= 0, "Cool down interval must not be negative.");
+        require(coolDownInterval > 0, Messages.INVALID_COOL_DOWN_INTERVAL);
         this.coolDownInterval = coolDownInterval;
         this.bePublic = bePublic;
 
-        creationTimestamp = Runtime.getTime();
-        lastUpdateTime = -1;
-        finishedCount = 0;
-        nextMilestoneIndex = 0;
-        paid = false;
-        finished = false;
+        this.creationTimestamp = Runtime.getTime();
```

Comment:
"this" is not really required here. I prefer to only use it when necessary but it's up to you :)

Reply:
I set my IDEA to add `this` to member access before saving the file, in case I accidentally create some local vars with the same name and break things silently.